snorkel/labeling/model/label_model.py

```diff
@@ -121,14 +121,14 @@ def _build_mask(self):
                     self.mask[si:ei, sj:ej] = 0
                     self.mask[sj:ej, si:ei] = 0
 
-    def _generate_O(self, L):
+    def _generate_O(self, L, higher_order=False):
         """Form the overlaps matrix, which is just all the different observed
         combinations of values of pairs of sources
 
         Note that we only include the k non-abstain values of each source,
         otherwise the model not minimal --> leads to singular matrix
         """
-        L_aug = self._get_augmented_label_matrix(L)
+        L_aug = self._get_augmented_label_matrix(L, higher_order=higher_order)
```

Comment:
why is this change happening here? is it just to facilitate testing?

Reply:
yes, facilitating testing + for later, we might make the O_inv method public and want to pass this argument in